### ISSUE TITLE
Undefined method `begin' for PseudoSourceRange

### ISSUE THREAD

**Issue by nononoy** [reviewdog/action-rubocop#55]
I'm getting an error during execution. Can provide more details if needed
```
undefined method `begin' for #<RuboCop::Cop::Offense::PseudoSourceRange:0x000055906792fa48>
Did you mean?  begin_pos
/home/runner/work/_actions/reviewdog/action-rubocop/v2/rdjson_formatter/rdjson_formatter.rb:41:in `build_diagnostic'
```
Workflow config:
```
jobs:
  rubocop:
    name: rubocop
    runs-on: ubuntu-latest
    steps:
      - name: Check out code
        uses: actions/checkout@v1
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: 2.7.1
      - name: rubocop
        uses: reviewdog/action-rubocop@v2
        with:
          rubocop_version: 0.90.0
          rubocop_extensions: rubocop-rails:gemfile rubocop-rspec:gemfile
          reporter: github-pr-review
          fail_on_error: true
```

**Comment by johnofsydney** [comment]
I had the same issue and swapped `uses: reviewdog/action-rubocop@v2` to `uses: reviewdog/action-rubocop@v1`

Everything seems to be working ok now

**Comment by ohbarye** [comment]
I guess the suggestion feature that is introduced since v2 is related to the error.

https://github.com/reviewdog/action-rubocop/blob/4c4064d4416a071633461de2549048c309d98d14/rdjson_formatter/rdjson_formatter.rb#L71-L72

@nononoy @johnofsydney @jeremybdk 
Could you give me an example Ruby code to reproduce the error?

**Comment by jeremybdk** [comment]
The thing is I don't really know what's causing this error, as it's always present, is there a way to have more details about the error?

**Comment by owst** [comment]
@ohbarye @jeremybdk I was able to reproduce in a simple way:
```
$ mkdir repro && cd repro
$ git clone git@github.com:reviewdog/action-rubocop.git
Cloning into 'action-rubocop'...
[...]
$ gem install rubocop
Successfully installed rubocop-1.23.0
[...]

$ touch empty_file.rb
$ rubocop empty_file.rb
Inspecting 1 file
W

Offenses:

empty_file.rb:1:1: W: Lint/EmptyFile: Empty file detected.

1 file inspected, 1 offense detected

$ rubocop --require ./action-rubocop/rdjson_formatter/rdjson_formatter.rb --format RdjsonFormatter empty_file.rb
{"source":{"name":"rubocop","url":"https://rubocop.org/"},"diagnostics":[]}
undefined method `begin' for #<RuboCop::Cop::Offense::PseudoSourceRange:0x00007f9dd5949ab0>
Did you mean?  begin_pos
/private/tmp/repro/action-rubocop/rdjson_formatter/rdjson_formatter.rb:41:in `build_diagnostic'
/private/tmp/repro/action-rubocop/rdjson_formatter/rdjson_formatter.rb:19:in `block in file_finished'
/private/tmp/repro/action-rubocop/rdjson_formatter/rdjson_formatter.rb:18:in `each'
/private/tmp/repro/action-rubocop/rdjson_formatter/rdjson_formatter.rb:18:in `file_finished'
[...]
```

I wonder if https://github.com/rubocop/rubocop/pull/8823 will give any insight to the fix - it sounds like it fixed a (very?) similar issue: https://github.com/rubocop/rubocop/issues/8718

**Comment by ohbarye** [comment]
Thanks @owst , I figured out the cause. Let me try to fix in #56.
